cEP-0026.md

```diff
@@ -0,0 +1,189 @@
+# Optimize caching for the NextGen-Core
+
+| Metadata |                                           |
+| ---------| ----------------------------------------- |
+| cEP      | 26                                        |
+| Version  | 1.0                                       |
+| Title    | Optimize caching for the NextGen-Core     |
+| Authors  | Palash Nigam <mailto:npalash25@gmail.com> |
+| Status   | Implementation Due                        |
+| Type     | Feature                                   |
+
+## Abstract
+
+This document describes how a caching mechanism (an enhanced version of the one
+being used by the old core) will be implemented and integrated with the
+NextGen-Core as a part of this
+[GSoC project](https://summerofcode.withgoogle.com/projects/#6434190552203264).
+
+## Introduction
+
+Since the NextGen-Core aims to provide faster speed and better performance for
+coala-runs on large codebases it needs caching to achieve this kind of
+performance.
+
+The current caching used by the old core can be improved upon and integrated
+with the NextGen-Core by adding a few more enhancements to the current caching
+mechanism like `FileFactory` class to load files, `Directory` class
```

Comment:
`FileFactory` & `Directory` are not mechanisms. Please rephrase it. 3rd party might not be aware of these classes before reading the details. Its not a good idea to put it in intro.

Reply:
You can just say extra classes/method to interface with files/dirs smoothly